packages/compartment-mapper/src/types.js

```diff
@@ -420,12 +423,84 @@ export {};
  */
 
 /**
- * @typedef {object} Attenuator
- * @property {Function} [attenuateGlobals]
- * @property {Function} [attenuateModule]
+ * @template {[any, ...any[]]} [GlobalParams=[any, ...any[]]]
+ * @template {[any, ...any[]]} [ModuleParams=[any, ...any[]]]
+ * @typedef Attenuator
+ * @property {GlobalAttenuatorFn<GlobalParams>} [attenuateGlobals]
+ * @property {ModuleAttenuatorFn<ModuleParams>} [attenuateModule]
+ */
+
+/**
+ * @template {[any, ...any[]]} [Params=[any, ...any[]]]
+ * @callback GlobalAttenuatorFn
+ * @param {Params} params
+ * @param {Record<PropertyKey, any>} originalObject
+ * @param {Record<PropertyKey, any>} globalThis
+ * @returns {void}
+ * @todo Unsure if we can do much typing of `originalObject` and `globalThis` here.
```

Comment:
`originalObject` is the same as the value passed in as `global` to `importLocation` options. I don't think it's reasonable to thread it all the way if TS can't infer.

Reply:
Unsure if there's something actionable here